docs/vagrant.md

```diff
@@ -0,0 +1,102 @@
+
+Although Vagrant supports [VirtualBox](https://www.virtualbox.org/), [Hyper-V](https://learn.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v), [Docker](https://www.docker.com/), or [VMware Fusion](https://customerconnect.vmware.com/downloads/get-download?downloadGroup=FUS-PUBTP-2021H1), `VirtualBox` is the only environment that has been developed and tested thus far. If you have need to use Hyper-V, Docker, or VMware, please create an issue and request a new feature.
+
+## Prerequisites
+
+1. You will require 4 GB of memory.
+1. You will require 8 GB of disk space.
+1. `npm install` creates symbolic links. As a result, the VM user: `vagrant` requires `create symbolic link` authority. In some environements, only the administrator has this authority. To resolve this issue in Windows, the following steps can be performed:
+   1. Launch the `Local Security Policy` app from the `start menu` (or `Win+R`, then type `secpol.msc`)
+   1. Navigate to `Local Policies` -> `User Rights Assignment`.
+   1. Open the `Create symbolic links` property. By default it has only `Administrators` listed.
+   1. Click `Add User or Group...` button and type `Authenticated Users` in the `Enter the object names to select` field and then press the `OK` button. This will grant `symbolic link` priviledge to all users who log on with credetials.
+   1. Close any open windows by clicking `OKs` and and then `log off` and `log on` to activate.
+   1. Symbolic links should now work in VirtualBox shared folders without administrator priviledges.
+
+## Setup Steps
+
+1. [Download](https://developer.hashicorp.com/vagrant/downloads?ajs_aid=ba208f95-d5a1-457d-abeb-49d458b95eec&product_intent=vagrant) Vagrant software.
+1. Reference these [installation](https://developer.hashicorp.com/vagrant/docs/installation) instructions.
+1. Verify installation by running `vagrant -v` in a terminal session.
+1. Install [VirtualBox](https://www.virtualbox.org/).
+1. If you have not already done so, fork repo to your local machine.
+1. Next you will need to install a [box](https://developer.hashicorp.com/vagrant/tutorials/getting-started/getting-started-boxes). The `Vagrantfile` in this repository uses box: `generic/ubuntu2204`. So to install this box, run in a terminal window:
+
+```bash
+    vagrant box add generic/ubuntu2204
+```
+
+To see which boxes are installed run:
+
+```bash
+    vagrant box list
+```
+
+7. Create a new folder and name it something that identifies the VM you are about to create. Example: `UbuntuVM`. The name of this folder will be used by Vagrant to identify this specific VM. It is to this directory that you will copy the vagrant file we have created for you. In a later step, Vagrant will create subdirectory`.vagrant` which it will use to manage your VM. It is in this directory where you will run all Vagrant commands that are specific to your VM.
```

Comment:
@arrocke It almost sounds to me like we should check `UbuntuVM` (or another name) into the git repo, and not bother with having a top-level `.vagrant` directory that people have to copy from.

Reply:
I think if we did this, we could remove point 10 and just have `HOST_FOLDER = "../"` in the `Vagrantfile`

Reply:
I see there is a disconnect in our conversation.  This has to do with how I setup my directory structure when learning and developing this code with respect to what you are visualizing.  Let me explain...

In my directory structure, UbuntuVM is in a parallel directory to the repo.  This is why I have `HOST_FOLDER = .../"gloss-translation"` as the host URL. 
![image](https://github.com/arrocke/gloss-translation/assets/141659916/a0183f19-1dd0-4ffa-a0d3-d476b287ac6b)
I believe you are visualizing the UbuntuVM folder within the gloss-translation repository. 
Is this correct?

If so, I believe this is a better approach and found out that this is the preferred method.

So I am suggesting the following changes to this PR:

***Revised Directory Structure***
![image](https://github.com/arrocke/gloss-translation/assets/141659916/bc046e72-fca4-426f-8b3a-37e91867e303)

***Notes:***
- I have added a sub-directory `virualbox` to folder `.vagrant`. This is where I have placed the Vagrantfile configured for a Virtualbox VM. This is because I have already developed a Vagrantfile for the Docker environment. I plan to add a new issue/PR for this new environment. At that time I plan to add another sub-directory under `.vagrant` called `docker` into which I will place the Vagrantfile with configuration to create a docker VM. 
![image](https://github.com/arrocke/gloss-translation/assets/141659916/a3543707-474d-4251-baa1-b7f27b891d6e)
- I have renamed directory `UbuntuVM` to `vm` in order to keep it more generic, just in case someone wants an O/S other than Ubuntu.
- I have updated the .gitignore file to exclude the contents of the `vm` directory.  This is because when the user runs `Vagrant up`, files will be created that are specific to that user's machine.  These should not be stored in the upstream repository. See excerpt below:

    ```text
    # Virtual Machine
    vm/*
    !vm/.gitkeep
    ```

Please let me know if you approve of this recommendation and if you would like to modify anything I have described above.  I will await your reply before proceeding any further.